### PR TITLE
Add User field to identity

### PR DESCRIPTION
**Pull request by mrajashree**
https://github.com/rancher/rancher/issues/8744
`User=true` will be set by all providers separately for all user accounts
`Me=true` will be set by cattle's listInternal, for all identities of the authenticatedAsAccount

### CODE REVIEW

**Comment by mrajashree** [comment]
This for loop https://github.com/rancher/cattle/pull/2690/commits/6401fad6dbaec8280d682cf9071b729e80584043#diff-d09c650621963e26b109b1edbdfa0658R71
sets `Me=true` on all identities of the current policy. The policy returns identities only for authenticatedAsAccount. So I think right now any more checks won't be needed and we can set `me=true` on all identities of the policy. 
@cjellick @prachidamle Let me know if this seems correct

File: code/framework/api/src/main/java/io/cattle/platform/api/auth/Identity.java [hunk 1]
```diff
@@ -19,6 +19,8 @@
     private String login = null;
     private String role = null;
     private String projectId = null;
+    private Boolean me = false;
```
Comment (ibuildthecloud) [inline code/framework/api/src/main/java/io/cattle/platform/api/auth/Identity.java]: Can you use `boolean` and not `Boolean`?  Is there a specific reason you used the boxed type?

File: resources/content/schema/user/user-auth.json [hunk 2]
```diff
@@ -300,6 +300,8 @@
         "identity.projectId": "cr",
         "identity.role": "cr",
         "identity.all": "cr",
+        "identity.me": "cr",
+        "identity.user": "cr",
```
Comment (prachidamle) [inline resources/content/schema/user/user-auth.json]: I think these can be only "r"

**Comment by prachidamle** [comment]
We should also change RancherIdentityProvider to add the "user=true" flag on the identities returned

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LDAPIdentityProvider.java [hunk 3]
```diff
@@ -194,6 +194,7 @@ protected Identity attributesToIdentity(LdapName dn){
             String accountName;
             String externalId = dn.toString();
             String login;
+            Boolean user = false;
```
Comment (cjellick) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LDAPIdentityProvider.java]: should be lowercase boolean

Comment (mrajashree) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LDAPIdentityProvider.java]: @cjellick ah yes, hadn't checked the PR again after updating it and this didn't show any errors in eclipse so didn't realize

File: code/framework/api/src/main/java/io/cattle/platform/api/auth/Identity.java (hunk 1 above)
Comment (mrajashree) [inline code/framework/api/src/main/java/io/cattle/platform/api/auth/Identity.java]: @ibuildthecloud changed it to `boolean`.

**Comment by cjellick** [comment]
Build failures are unrelated. Saw them in other merge builds

**Comment by cjellick** [comment]
LGTM

**Pull request merged**